File: 2240-number-of-ways-to-buy-pens-and-pencils/2240-number-of-ways-to-buy-pens-and-pencils.cpp
class Solution {
public:
    long long waysToBuyPensPencils(int total, int cost1, int cost2) 
    {
        long long maxi = max(cost1, cost2);
        long long mini = min(cost1, cost2);
        if(mini>total)
            return 1;
        long long s=0, remaining;
        for(long long i=0; i<=(total/maxi); i++)
        {
            remaining = total - (maxi*i);
            s += (remaining/mini)+1;
        }
        return s;
    }
};

File: 2240-number-of-ways-to-buy-pens-and-pencils/2240-number-of-ways-to-buy-pens-and-pencils.java
class Solution {
    public long waysToBuyPensPencils(int total, int cost1, int cost2) 
    {
        long s=0, max, min, lim;
        max = Math.max(cost1, cost2);
        min = Math.min(cost1, cost2);
        if(min>total)
            return 1;
        lim = total/max;
        for(long i=0; i<=lim; i++)
            s += ((total-(max*i))/min)+1;
        return s;    
    }
}